### PR TITLE
Store dimension hidden option in the metadata db

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
Store the `hidden` option of a dimension (e.g.  cpu idle) in the database. This is to ensure that queries on
archived charts (in a following PR) will correctly skip hidden dimensions unless explicitly requested.

##### Test Plan
- After applying this PR the `options` column in the `dimension` table in `/var/cache/netdata/netdata-meta.db` will have the value `hidden` for the appropriate dimensions. 
   - Example: 
     ```
     select distinct name, options from dimension where options is not null;
     name  options
     ----  -------
     idle  hidden 
     ```

### CODE REVIEW

File: database/sqlite/sqlite_functions.c [hunk 1]
```diff
@@ -905,6 +905,49 @@ int sql_store_dimension(
...
+    int rc;
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE)
+            return 0;
+        error_report("Database has not been initialized");
+        return 1;
+    }
+
+    rc = sqlite3_prepare_v2(db_meta, "UPDATE dimension SET options = @options WHERE dim_id = @dim_id", -1, &res, 0);
```
Comment (vkalintiris) [inline database/sqlite/sqlite_functions.c:923]: - How are we going to serialize/deserialize options if we need to save multiple of them in the future?
- Is storing the raw rrdim options a better/simpler approach? We are a long way from exhausting the available bits.

Comment (stelfrag) [reply database/sqlite/sqlite_functions.c:923]: Its good point but currently we don't have a use case I can think of that needs to persist any other option. If needed we can easily migrate in the future to store raw rrdim options

**Review by vkalintiris — APPROVED** [review]
LGTM!

**Review by thiagoftsm — APPROVED** [review]
The PR  worked as expected:

```sh
metadata_migration                                     
node_instance                                          
v_chart_hash  
sqlite> select distinct name, options from dimension where options is not null;
idle|hidden
```

@stelfrag , please, update the description with correct path to database (`/var/cache/netdata/netdata-meta.db`)

**Pull request merged**